### PR TITLE
Draft of LLVM AMD-GPU

### PR DESCRIPTION
**Pull request by Micket**
@jpsamaroo I wasn't sure what targets that needed to be built, or what products we care about here. 

I also wasn't sure what the heck to call this package.

### CODE REVIEW

**Comment by vchuravy** [comment]
I would sort this as: `R/ROCM/LLVM` and call the package itself `rocm-llvm`

**Comment by giordano** [comment]
Note that the directory where the builder is must match exactly the name.  Also, no dashes allowed.  So maybe `R/ROCM/rocm_llvm`?

File: L/LLVM_AMDGPU/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# The rest of the ROCm stack only seem to support 64bit Linux:
+platforms = [
+    Linux(:x86_64, libc=:glibc),
+    Linux(:x86_64, libc=:musl),
```
Comment (giordano) [inline L/LLVM_AMDGPU/build_tarballs.jl:35]: Same comment as https://github.com/JuliaPackaging/Yggdrasil/pull/1735#discussion_r499186246
```suggestion
    Platform("x86_64", "linux"; libc="glibc"),
    Platform("x86_64", "linux"; libc="musl"),
```

File: L/LLVM_AMDGPU/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    ExecutableProduct("opt", :opt, "tools"),
+    ExecutableProduct("llc", :llc, "tools"),
+    ExecutableProduct("llvm-mca", :llvm_mca, "tools"),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("Zlib_jll"),
+    Dependency("Ncurses_jll"), # 
+    BuildDependency("Python_jll"),
```
Comment (giordano) [inline L/LLVM_AMDGPU/build_tarballs.jl:53]: Why do you need Python as build dependency?  Do you need to run Python scripts during the build?  Python should be already available.  Also, this dependency will install the package for the target platform, if you need to run the Python interpreter that won't generally work

File: L/LLVM_AMDGPU/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies,
+               preferred_gcc_version=v"8")
```
Comment (giordano) [inline L/LLVM_AMDGPU/build_tarballs.jl:58]: Any particular reason to use GCC 8?

**Comment by Micket** [comment]
> I would sort this as: `R/ROCM/LLVM` and call the package itself `rocm-llvm`

> Note that the directory where the builder is must match exactly the name. Also, no dashes allowed. So maybe `R/ROCM/rocm_llvm`?

hsakmt_roct and hsa_rocr are also things from the ROCm "bundle". Shall we also bring those in under the ROCM directory?

**Comment by giordano** [comment]
> hsakmt_roct and hsa_rocr are also things from the ROCm "bundle". Shall we also bring those in under the ROCM directory?

No problem with me, but don't do it in this PR.  If you want to move them, remember to add `[skip ci]` to the commit message so merging the PR won't trigger a rebuild & registration.  Also, always remember that the directory must match the name of the builder

**Comment by jpsamaroo** [comment]
I think you want `-DLLVM_BUILD_LLVM_DYLIB:BOOL=OFF -DLLVM_LINK_LLVM_DYLIB:BOOL=OFF` instead of `-DBUILD_SHARED_LIBS=OFF`, although maybe it works the same either way.

File: L/LLVM_AMDGPU/build_tarballs.jl (hunk 3 above)
Comment (Micket) [reply L/LLVM_AMDGPU/build_tarballs.jl]: No, probably just copy paste. Dropped.

File: L/LLVM_AMDGPU/build_tarballs.jl (hunk 2 above)
Comment (Micket) [reply L/LLVM_AMDGPU/build_tarballs.jl]: I based this list blindly on the Spack recipe
https://github.com/RadeonOpenCompute/rocm-spack/blob/df2c988a4aa1645c72b2f9164cd2a8907d3dc410/var/spack/repos/builtin/packages/llvm-amdgpu/package.py
I'll test dropping the python dep

**Comment by jpsamaroo** [comment]
@Micket I've been working on building all of the ROCm components using the regular LLVM shlib, which seems to generally work well (although it has some slight issues on musl that I'm working around). It seems that we don't need AMD's LLVM fork for any of their libraries, so I think if you're ok with that, we can close this PR.

**Comment by Micket** [comment]
I'll admit I kind of gave up on this as there didn't seem to exist any support for me RDNA card.

**Pull request closed without merging**